// ngx-lottie/esm2015/src/lottie.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: src/lottie.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ChangeDetectionStrategy, Inject, ElementRef, ViewChild, Self, Renderer2, PLATFORM_ID } from '@angular/core';
import { BaseDirective } from './base.directive';
import { AnimationLoader } from './animation-loader';
import { LottieEventsFacade } from './events-facade';
export class LottieComponent extends BaseDirective {
    /**
     * @param {?} renderer
     * @param {?} platformId
     * @param {?} eventsFacade
     * @param {?} animationLoader
     */
    constructor(renderer, platformId, eventsFacade, animationLoader) {
        super(platformId, animationLoader);
        this.renderer = renderer;
        this.eventsFacade = eventsFacade;
        this.container = (/** @type {?} */ (null));
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        /** @type {?} */
        const containerClass = changes.containerClass;
        if (containerClass === undefined || typeof containerClass.currentValue !== 'string') {
            return;
        }
        this.renderer.addClass(this.container.nativeElement, containerClass.currentValue);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        super.setWidthAndHeight();
        super.loadAnimation(this.container.nativeElement, this.eventsFacade, this);
    }
}
LottieComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng-lottie',
                template: `
    <div #container [style.width]="width" [style.height]="height" [ngStyle]="styles"></div>
  `,
                changeDetection: ChangeDetectionStrategy.OnPush,
                providers: [LottieEventsFacade]
            }] }
];
/** @nocollapse */
LottieComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: String, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] },
    { type: LottieEventsFacade, decorators: [{ type: Self }] },
    { type: AnimationLoader }
];
LottieComponent.propDecorators = {
    container: [{ type: ViewChild, args: ['container', { static: true },] }]
};
if (false) {
    /** @type {?} */
    LottieComponent.prototype.container;
    /**
     * @type {?}
     * @private
     */
    LottieComponent.prototype.renderer;
    /**
     * @type {?}
     * @private
     */
    LottieComponent.prototype.eventsFacade;
}
